File: src/main/java/kome/hatsuneakiko/pastenbt/ModConfig.java
package kome.hatsuneakiko.pastenbt;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import net.minecraft.server.level.ServerPlayer;
import net.minecraftforge.fml.loading.FMLPaths;

import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.util.ArrayList;
import java.util.List;

public class ModConfig {
    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
    private static final File CONFIG_FILE = FMLPaths.CONFIGDIR.get().resolve("nbtpaster.json").toFile();
    
    public static Data INSTANCE = new Data();

    public static class Data {
        public List<String> allowedPlayers = new ArrayList<>();
        public List<String> allowedUUIDs = new ArrayList<>();
        public boolean allowOps = true;
        public int blocksPerTick = 100;
        public int maxActiveTasks = 2;
        public int maxBlocksPerTask = 200000;
        public boolean skipAir = true;
    }

    public static boolean load() {
        if (!CONFIG_FILE.exists()) {
            save();
            return true;
        }
        try (FileReader reader = new FileReader(CONFIG_FILE)) {
            Data data = GSON.fromJson(reader, Data.class);
            if (data != null) INSTANCE = data;
            return true;
        } catch (Exception e) {
            PasteNbtMod.LOGGER.error("Failed to load config", e);
            return false;
        }
    }

    public static void save() {
        try (FileWriter writer = new FileWriter(CONFIG_FILE)) {
            GSON.toJson(INSTANCE, writer);
        } catch (Exception e) {
            PasteNbtMod.LOGGER.error("Failed to save config", e);
        }
    }

    public static boolean canUse(ServerPlayer player) {
        if (INSTANCE.allowOps && player.hasPermissions(2)) return true;
        if (INSTANCE.allowedPlayers.contains(player.getGameProfile().getName())) return true;
        return INSTANCE.allowedUUIDs.contains(player.getUUID().toString());
    }
}

File: src/main/java/kome/hatsuneakiko/pastenbt/ModConfig.kt
package kome.hatsuneakiko.pastenbt

import com.google.gson.GsonBuilder
import net.minecraft.server.level.ServerPlayer
import net.minecraftforge.fml.loading.FMLPaths
import java.io.File
import java.io.FileReader
import java.io.FileWriter

object ModConfig {
    private val GSON = GsonBuilder().setPrettyPrinting().create()
    private val CONFIG_FILE = FMLPaths.CONFIGDIR.get().resolve("nbtpaster.json").toFile()
    
    var INSTANCE = Data()

    class Data {
        val allowedPlayers = mutableListOf<String>()
        val allowedUUIDs = mutableListOf<String>()
        var allowOps = true
        var blocksPerTick = 100
        var maxActiveTasks = 2
        var maxBlocksPerTask = 200000
        var skipAir = true
    }

    fun load(): Boolean {
        if (!CONFIG_FILE.exists()) {
            save()
            return true
        }
        return try {
            FileReader(CONFIG_FILE).use { reader ->
                INSTANCE = GSON.fromJson(reader, Data::class.java) ?: Data()
            }
            true
        } catch (e: Exception) {
            PasteNbtMod.LOGGER.error("Failed to load config", e)
            false
        }
    }

    fun save() {
        try {
            FileWriter(CONFIG_FILE).use { writer ->
                GSON.toJson(INSTANCE, writer)
            }
        } catch (e: Exception) {
            PasteNbtMod.LOGGER.error("Failed to save config", e)
        }
    }

    fun canUse(player: ServerPlayer): Boolean {
        if (INSTANCE.allowOps && player.hasPermissions(2)) return true
        if (INSTANCE.allowedPlayers.contains(player.gameProfile.name)) return true
        return INSTANCE.allowedUUIDs.contains(player.uuid.toString())
    }
}
